src/position.rs

```diff
@@ -23,6 +23,22 @@ pub struct RawSpan {
     pub end: ByteIndex,
 }
 
+impl RawSpan {
+    /// Fuse two spans if they are from the same source file. The resulting span is the smallest
+    /// span that contain both `span1` and `span2`.
+    pub fn fuse(span1: RawSpan, span2: RawSpan) -> Option<RawSpan> {
+        if span1.src_id == span2.src_id {
+            Some(RawSpan {
+                src_id: span1.src_id,
```

Comment:
What is the `src_id`?

Reply:
It's a identifier for the source file (differentiating e.g. positions in "foo.ncl" and "bar.ncl").